pkg/controller/atlasproject/cloud_provider_integration.go

```diff
@@ -0,0 +1,385 @@
+package atlasproject
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/status"
```

Comment:
same as above, to denote that this is a "Atlas Kubernetes v2.0 status package" I suggest to call this `akov2status`. This is also in alignment with kube standards.